EssentialFeed/EssentialFeedTests/Image Comments/API/ImageCommentsMapperTests.swift

```diff
@@ -0,0 +1,75 @@
+//
+// Copyright © 2021 Essential Developer. All rights reserved.
+//
+
+import XCTest
+import EssentialFeed
+
+class ImageCommentsMapperTests: XCTestCase {
+	func test_map_throwsErrorOnNon200HTTPResponse() throws {
+		let json = makeItemsJSON([])
+		let samples = [199, 300, 400, 500]
+
+		try samples.forEach { code in
+			XCTAssertThrowsError(
+				try ImageCommentsMapper.map(json, from: HTTPURLResponse(statusCode: code))
+			)
+		}
+	}
+
+	func test_map_throwsErrorOn200HTTPResponseWithInvalidJSON() throws {
+		let invalidJSON = Data("invalid json".utf8)
+
+		let samples = [199, 300, 400, 500]
+
+		try samples.forEach { sample in
+			XCTAssertThrowsError(
+				try ImageCommentsMapper.map(invalidJSON, from: HTTPURLResponse(statusCode: sample))
+			)
+		}
+	}
+
+	func test_map_deliversNoItemsOn200HTTPResponseWithEmptyJSONList() throws {
```

Comment:
Since the comments api considers all 2xx codes to be valid then it is better to test not only 200 but also some portion of other 2xx codes.
Please fix the name of the test and add more samples
```suggestion
	func test_map_deliversNoItemsOn2xxHTTPResponseWithEmptyJSONList() throws {
```

Reply:
Renamed.

Reply:
Hi @tiagoBAmorim!
I suggested only the fix the name of the test part. The other one add more samples should be implemented by you without my hints. Please add assertions for other valid codes, not only for 200 at least for 201, 249 and for 299

Reply:
I think I got it this time :)